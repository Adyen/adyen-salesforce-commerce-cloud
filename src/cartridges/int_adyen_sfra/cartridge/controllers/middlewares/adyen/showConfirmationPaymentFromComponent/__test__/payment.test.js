const OrderMgr = require('dw/order/OrderMgr');
const URLUtils = require('dw/web/URLUtils');
const adyenCheckout = require('*/cartridge/scripts/adyenCheckout');
const handlePayment = require('../payment');

let res;
let next;
let req;
let order;
beforeEach(() => {
  jest.clearAllMocks();
  req = { locale: { id: 'mocked_locale' } };
  res = { redirect: jest.fn() };
  order = OrderMgr.getOrder();
  next = jest.fn();
});

const paymentErrorSnap = `
  Array [
    Array [
      "Checkout-Begin",
      "stage",
      "placeOrder",
      "paymentError",
      "mocked_error.payment.not.valid",
    ],
  ]
`;

describe('Payment', () => {
  it('should successfully handle payment', () => {
    adyenCheckout.doPaymentsDetailsCall.mockReturnValue({
      resultCode: 'Authorised',
    });
    const stateData = {
      paymentData: 'mocked_paymentData',
      details: 'mocked_details',
    };
    handlePayment(stateData, order, { req, res, next });
    expect(res.redirect.mock.calls).toMatchSnapshot();
  });

  it('should handle payment error when not Authorised', () => {
    adyenCheckout.doPaymentsDetailsCall.mockReturnValue({
      resultCode: 'Not_Authorised',
    });
    const stateData = {
      paymentData: 'mocked_paymentData',
      details: 'mocked_details',
    };
    handlePayment(stateData, order, { req, res, next });
    expect(URLUtils.url.mock.calls).toMatchInlineSnapshot(paymentErrorSnap);
  });

  it('should handle payment error when theres not state data', () => {
    const stateData = {};
    handlePayment(stateData, order, { req, res, next });
    expect(URLUtils.url.mock.calls).toMatchInlineSnapshot(paymentErrorSnap);
  });
});
