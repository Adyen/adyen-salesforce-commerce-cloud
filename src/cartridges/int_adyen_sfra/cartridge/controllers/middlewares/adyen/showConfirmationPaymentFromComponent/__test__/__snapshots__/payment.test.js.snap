// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment should successfully handle payment 1`] = `
Array [
  Array [
    "[\\"Order-Confirm\\",\\"ID\\",\\"mocked_orderNo\\",\\"token\\",\\"mocked_orderToken\\"]",
  ],
]
`;
